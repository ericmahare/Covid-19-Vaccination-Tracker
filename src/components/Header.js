import React from 'react'
import Hero from './Hero'
import { Link } from "react-router-dom";

const Header = ({ img, vaccinated, country, total }) => {
  return (
    <header>
      <nav>
        <ul className='nav-list'>
          <li>{country ? <li className='back'><Link to='/'>{`<`}</Link></li> : '2022'}</li>
          <li>Covid-19 Vaccination</li>
          <li>
            <i className="fa-solid fa-microphone"></i>
            <i className="fa-solid fa-gear"></i>
          </li>
        </ul>
      </nav>
      <Hero map={img} country={country} vaccinated={vaccinated} total={total}/>
    </header>
  )
}

export default Header