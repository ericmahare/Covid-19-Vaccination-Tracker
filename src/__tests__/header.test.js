import renderer from 'react-test-renderer';
import Header from '../components/Header';
import { BrowserRouter as Router } from 'react-router-dom';

const img = './images/africa.svg';
const vaccinated = 30000;
const country = 'Kenya';
const total = 70000;

it('renders correctly', () => {
  const tree = renderer.create(
    <Router>
      <Header img={img} vaccinated={vaccinated} country={country} total={total}/>
    </Router>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})