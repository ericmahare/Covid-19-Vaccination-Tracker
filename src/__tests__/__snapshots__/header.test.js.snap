// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header>
  <nav>
    <ul
      className="nav-list"
    >
      <li>
        <li
          className="back"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            &lt;
          </a>
        </li>
      </li>
      <li>
        Covid-19 Vaccination
      </li>
      <li>
        <i
          className="fa-solid fa-microphone"
        />
        <i
          className="fa-solid fa-gear"
        />
      </li>
    </ul>
  </nav>
  <div
    className="hero"
  >
    <div
      className="hero-img-container"
    >
      <img
        alt="Africa"
        className="flag"
        src="./images/africa.svg"
      />
    </div>
    <div
      className="hero-info"
    >
      <h2>
        Kenya
      </h2>
      <h2>
        (30,000) people vaccinated
      </h2>
    </div>
  </div>
</header>
`;
